snippets/card-product.liquid

```diff
@@ -118,7 +118,33 @@
               </a>
             </h3>
           </div>
-          <div class="card__badge {{ settings.badge_position }}">
+
+          {% assign CustomTags = settings.custom_badges  | newline_to_br | split: '<br />' %}
+          {% assign tagLabel = 0 %}
+          {% assign hasTag = 0 %}
+
+          {% for tag in CustomTags %}
+            {% assign tagSplit = tag | split: ':' %}
+            {% assign pTag = tagSplit.first %}
+            {% assign pLabel = tagSplit.last %}
+
+            {% assign hasTag = card_product.tags | where: pTag %}
+
+            {% if hasTag.size > 0 %}
+              {% assign tagLabel = pLabel %}
+            {% endif %}
+          {% endfor %}
+
+          <div class="card__badge {{ settings.badge_position }}" {% if tagLabel != 0 %} data-tag="{{ tagLabel }}"{% endif %}>
+            {% if tagLabel != 0 %}
+              <span
+                class="badge custom-badge badge--bottom-left color-{{ settings.sale_badge_color_scheme }}"
+                style="--color-background: {{ settings.custom_badge_color_bg.red }}, {{ settings.custom_badge_color_bg.green }}, {{ settings.custom_badge_color_bg.blue }};"
+              >
+                {{- tagLabel -}}
+              </span>
+            {% endif %}
+
```

Comment:
custom badge colors not needed here either, just re-use the `sale_badge_color_scheme`

Reply:
We should keep with the stock badge. All we need is an additional `{%- if card_product.type == 'Wholesale' -%}` condition in this if/else block where it generates the badge text.